oci/pack.go

```diff
@@ -520,6 +521,44 @@ unpack:
 	return nil
 }
 
+// Delete deletes OCI package from the host machine.
+func (ocipack *ociPackage) Delete(ctx context.Context, version string) error {
+	ociDir := path.Join(config.G[config.KraftKit](ctx).RuntimeDir, "oci")
+
+	// Removing config layer
+	typeAndId := strings.Split(string(ocipack.image.manifest.Config.Digest), ":")
+	if err := os.RemoveAll(path.Join(ociDir, "configs", typeAndId[0], typeAndId[1])); err != nil {
+		return err
+	}
+
+	// Removing image layers
+	for _, layer := range ocipack.image.manifest.Layers {
+		typeAndId = strings.Split(string(layer.Digest), ":")
+		if err := os.RemoveAll(path.Join(ociDir, "layers", typeAndId[0], typeAndId[1])); err != nil {
+			return err
+		}
+	}
+
+	// Removing manifest file
+	if err := os.RemoveAll(path.Join(ociDir,
+		"manifests",
+		ocipack.Name(),
+		ocipack.Version()+".json")); err != nil {
+		return err
+	}
+
+	files, err := os.ReadDir(path.Join(ociDir, "manifests", ocipack.Name()))
+	if err != nil {
+		return err
+	}
+	if len(files) == 0 {
+		if err = os.RemoveAll(path.Join(ociDir, "manifests", ocipack.Name())); err != nil {
+			return err
+		}
+	}
```

Comment:
I think that the use of `ocipack.Name()` is incorrect here. My files were not deleted completely:

```
├── configs
│   └── sha256
├── layers
│   └── sha256
└── manifests
    ├── index.unikraft.io
    │   └── unikraft.org
    │       └── nginx
    │           └── latest.json
    └── unikraft.org
```

Reply:
@craciunoiuc Now give it a try, Hoping will work fine :)